### PR TITLE
Extend CLI interface to load form instances

### PR DESCRIPTION
**Pull request by phillipm**
Enable JavaRosa form runner CLI to load particular form instances
Try it out with: 
```
java -jar form_translate.jar runinstance form.xml instance.xml
```

### CODE REVIEW

File: util/schema-gen/src/org/javarosa/xform/schema/FormInstanceLoader.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Logic for loading a particular xml form instance into a FormDef.
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public class FormInstanceLoader {
+    private static InstanceInitializationFactory iif;
```
Comment (ctsims) [inline util/schema-gen/src/org/javarosa/xform/schema/FormInstanceLoader.java]: How does this get set?

We'll need to be able to define this for an XForm that just got parsed to be properly initialized.

**Comment by ctsims** [comment]
@phillipm looks good other than this one area.

File: util/schema-gen/src/org/javarosa/xform/schema/FormInstanceLoader.java (hunk 1 above)
Comment (phillipm) [inline util/schema-gen/src/org/javarosa/xform/schema/FormInstanceLoader.java]: No idea, it was unset in FormLoaderTask: https://github.com/dimagi/commcare-odk/blob/master/app/src/org/odk/collect/android/tasks/FormLoaderTask.java#L170

It looks like in this particular case the XFormPlayer already does the iif setup by calling XFormEnvironment.setup() and using https://github.com/dimagi/javarosa/blob/master/util/schema-gen/src/org/javarosa/engine/MockupProviderFactory.java

Comment (ctsims) [inline util/schema-gen/src/org/javarosa/xform/schema/FormInstanceLoader.java]: This variable is currently unused in this setup, then. It's private here and unset.

**Comment by phillipm** [comment]
@ctsims I removed the InstanceInitializationFactory since FormDef.initialize is called later anyways by the XFormPlayer.
Ready to look at again.

**Pull request merged**